PcAtChipsetPkg/PcatRealTimeClockRuntimeDxe/PcRtc.c

```diff
@@ -305,7 +305,11 @@ PcRtcInit (
   //
   // Clear RTC register D
   //
-  RegisterD.Data = FixedPcdGet8 (PcdInitialValueRtcRegisterD);
+  // On most platforms VRT is read-only, but when it's not it must
+  // be preserved to prevent failing in RtcWaitToUpdate().
+  //
+  RegisterD.Data = RtcRead (RTC_ADDRESS_REGISTER_D);
+  RegisterD.Data = (FixedPcdGet8 (PcdInitialValueRtcRegisterD) & 0x7f)| (RegisterD.Data & BIT7);

```

Comment:
I checked the PCH spec (https://www.intel.com/content/dam/www/public/us/en/documents/datasheets/9-series-chipset-pch-datasheet.pdf) and it has following explanation of VRT bit in Register D:

Valid RAM and Time Bit VRT - R/W.
0 - This bit should always be written as a 0 for write cycle, however it will return a 1 for read cycles.
1 - This bit is hardwired to 1 in the RTC power well.


So that means on a 1 is written with your changes.
That's a behavior change!

Reply:
Thanks for the review. The old ICH and PCH datasheets seem to be the only source for RTC register 0xd. It's not defined on newer Intel platforms.
To not break the older platforms I'll find another way of fixing this for recent AMD chipsets.